src/main/java/seedu/duke/ui/Ui.java

```diff
@@ -0,0 +1,39 @@
+package seedu.duke.ui;
+
+import java.util.Scanner;
+
+public class Ui {
+    private final Scanner scanner;
+
+    /**
+     * Constructs a UI instance with a Scanner for user input.
+     */
+    public Ui() {
+        scanner = new Scanner(System.in);
+    }
+
+    public String receiveUserInput() {
+        return scanner.nextLine();
+    }
+
+    public void showWelcome() {
+        System.out.println(UserOutput.WELCOME_MESSAGE.message);
+    }
+
+    public void showGoodbye() {
+        System.out.println(UserOutput.GOODBYE_MESSAGE.message);
+    }
+
+    /**
```

Comment:
I feel that this is not necessary for now.... in my opinion when we start implementing error handling we then include the necessary code if not we might forget about this method in the future so I feel that you can remove this for now :/